org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/IterableExtensions.java

```diff
@@ -323,12 +324,39 @@
 	}
 
 	/**
+	 * Returns an iterable that performs the given {@code transformation} for each element of {@code original} when
+	 * requested. The mapping is done lazily. That is, subsequent iterations of the elements in the iterable will
+	 * repeatedly apply the transformation.
+	 * <p>
+	 * The transformation maps each element to an iterable, and all resulting iterables are combined to a single iterable.
+	 * Effectively a combination of {@link #map(Iterable, Function1)} and {@link #flatten(Iterable)} is performed.
+	 * </p>
+	 * <p>
+	 * The returned iterable's iterator supports {@code remove()} if the provided iterator does. After a successful
+	 * {@code remove()} call, {@code original} no longer contains the corresponding element.
+	 * </p>
+	 * 
+	 * @param original
+	 *            the original iterable. May not be <code>null</code>.
+	 * @param transformation
+	 *            the transformation. May not be <code>null</code>.
+	 * @return an iterable that provides the result of the transformation. Never <code>null</code>.
+	 * 
+	 * @since 2.13
+	 */
+	@Pure
+	public static <T, R> Iterable<R> flatMap(Iterable<T> original, Function1<? super T, ? extends Iterable<R>> transformation) {
```

Comment:
I think the result type should be identical to `map()`, and that also returns `Iterable<R>`.